enginetest/script_queries.go

```diff
@@ -700,6 +700,34 @@ var ScriptTests = []ScriptTest{
 			{3, 3},
 		},
 	},
+	{
+		Name: "Indexed Join On Keyless Table",
+		SetUpScript: []string{
+			"create table l (pk int primary key, c0 int, c1 int);",
+			"create table r (c0 int, c1 int);",
+			"create index r_c0 on r (c0);",
+			"create index r_c1 on r (c1);",
+			"insert into l values (0, 0, 0), (1, 0, 1), (2, 1, 0), (3, 0, 2), (4, 2, 0), (5, 1, 2), (6, 2, 1), (7, 2, 2);",
+			"insert into r values (1, 1), (2, 2), (2, 2);",
+		},
+		Query: "select pk, l.c0, l.c1 from l join r on l.c0 = r.c0 or l.c1 = r.c1 order by 1, 2, 3;",
```

Comment:
Would be nice to have a test of n=3 join columns here to ensure that recursive OR logic works correctly